modules/core/include/opencv2/core/quaternion.inl.hpp

```diff
@@ -865,7 +865,155 @@ Quat<T> Quat<T>::spline(const Quat<T> &q0, const Quat<T> &q1, const Quat<T> &q2,
+        {R(1, 2), -R(0, 1), R(0, 0), -CV_PI * 0.5f , R(0, 1), R(0, 0), CV_PI * 0.5f, R(0, 2), R(2, 2),-R(1, 2), R(1, 0), R(1, 1), 0}, //INT_YXZ
+        {R(1, 0), R(0, 2), R(2, 2), CV_PI * 0.5f, R(0, 2), R(0, 1), -CV_PI * 0.5f, -R(2, 0), R(0, 0), R(1, 0), -R(1, 2), R(1, 1), 0}, //INT_YZX
+        {R(2, 1), R(1, 0), R(0, 0), CV_PI * 0.5f,R(1, 0), R(0, 0), -CV_PI * 0.5f, -R(0, 1), R(1, 1),R(2, 1),-R(2, 0), R(2, 2), 0},//INT_ZXY
+        {R(2, 0), -R(0, 1), R(1, 1), -CV_PI * 0.5f,R(1, 2), R(1, 1), CV_PI * 0.5f,R(1, 0), R(0, 0),-R(2, 0),R(2, 1), R(2, 2), 0},//INT_ZYX
+        {R(0, 0), R(2, 1), R(2, 2), 0, R(1, 2), R(1, 1), CV_PI, R(1, 0), -R(2, 0), R(0, 0), R(0, 1), R(0, 2), 0}, //INT_XYX
+        {R(0, 0), R(2, 1), R(2, 2), 0,-R(2, 1), R(2, 2), CV_PI, R(2, 0), R(1, 0), R(0, 0), R(0, 2), -R(0, 1), 0},//INT_XZX
+        {R(1, 1), R(0, 2), R(0, 0), 0 ,-R(2, 0), R(0, 0), CV_PI, R(0, 1), R(2, 1), R(1, 1), R(1, 0),-R(1, 2), 0},  //INT_YXY
+        {R(1, 1), R(0, 2), R(0, 0),0 ,R(0, 2), -R(0, 0) ,CV_PI ,R(2, 1), -R(0, 1) ,R(1, 1) ,R(1, 2), R(1, 0), 0}, //INT_YZY
+        {R(2, 2), R(1, 0), R(1, 1), 0, R(1, 0), R(0, 0), CV_PI, R(0, 2), -R(1, 2),R(2, 2), R(2, 0), R(2, 1), 0}, //INT_ZXZ
+        {R(2, 2), R(1, 0), R(0, 0), 0, R(1, 0), R(0, 0), CV_PI, R(1, 2), R(0, 2),R(2, 2), R(2, 1), -R(2, 0), 0}, //INT_ZYZ
+
+        {R(2, 0), -CV_PI * 0.5f, -R(0, 1), R(1, 1), CV_PI * 0.5f, R(1, 2), R(1, 1), R(2, 1), R(2, 2), -R(2, 0), R(1, 0), R(0, 0), 1}, //EXT_XYZ
+        {R(1, 0), CV_PI * 0.5f, R(0, 2), R(2, 2), -CV_PI * 0.5f, R(0, 2), R(0, 1), -R(1, 2), R(1, 1), R(1, 0), -R(2, 0), R(0, 0), 1}, //EXT_XZY
+        {R(2, 1), CV_PI * 0.5f, R(1, 0), R(0, 0), -CV_PI * 0.5f, R(1, 0), R(0, 0), -R(2, 0), R(2, 2),R(2, 1), -R(0, 1), R(1, 1), 1}, //EXT_YXZcd
+        {R(0, 2), -CV_PI * 0.5f, -R(1, 2), R(2, 2), CV_PI * 0.5f, R(2, 0), R(2, 2), R(0, 2), R(0, 0),-R(0, 1), R(2, 1), R(1, 1), 1}, //EXT_YZX
+        {R(1, 2), -CV_PI * 0.5f, -R(0, 1), R(0, 0), CV_PI * 0.5f, R(0, 1), R(0, 0), R(1, 0), R(1, 1),-R(1, 2), R(0, 2), R(2, 2), 1}, //EXT_ZXY
+        {R(0, 2), CV_PI * 0.5f, R(1, 0), R(1, 1), -CV_PI * 0.5f, R(2, 1), R(1, 1), -R(0, 1), R(0, 0),R(0, 2), -R(1, 2), R(2, 2), 1}, //EXT_ZYX
+        {R(0, 0), 0, R(2, 1), R(2, 2), CV_PI, R(1, 2), R(1, 1),R(0, 1), R(0, 2), R(0, 0),R(1, 0), -R(2, 0), 1}, //EXT_XYX
+        {R(0, 0), 0, R(2, 1), R(2, 2), CV_PI, R(2, 1), R(2, 2),R(0, 2), -R(0, 1), R(0, 0),R(2, 0), R(1, 0), 1}, //EXT_XZX
+        {R(1, 1), 0, R(0, 2), R(0, 0), CV_PI, -R(2, 0), R(0, 0),R(1, 0), -R(1, 2), R(1, 1),R(0, 1), R(2, 1), 1}, //EXT_YXY
+        {R(1, 1), 0 ,R(0, 2), R(0, 0), CV_PI, R(0, 2), -R(0, 0),R(1, 2), R(1, 0),R(1, 1), R(2, 1), -R(0, 1), 1}, //EXT_YZY
+        {R(2, 2), 0, R(1, 0), R(1, 1), CV_PI, R(1, 0), R(0, 0),R(2, 0), R(2, 1), R(2, 2),R(0, 2), -R(1, 2), 1}, //EXT_ZXZ
+        {R(2, 2), 0, R(1, 0), R(0, 0), CV_PI, R(1, 0), R(0, 0),R(2, 1), -R(2, 0), R(2, 2),R(1, 2), R(0, 2), 1} //EXT_ZYZ
+    };
```

Comment:
At each call to `toEulerAngles()` this table will be recalculated (or, to be exact, combined from `R` elements) from scratch. I'm not sure if a compiler would optimize this table away.
Wouldn't it be better to keep the indices in `R` matrix instead? Like `3*row + col`. This would make the code a bit more complicated, though, but it's reasonable anyway, I think.

Reply:
OK, I will use the indices of R matrix  in the array `rotationR`.